include/ck_tile/ops/epilogue/chainer/epilogue_chainer.hpp

```diff
@@ -0,0 +1,214 @@
+                                   Context& context) const
+    {
+        ck_tile::apply(
+            [&](auto&&... epilogue_args) {
+                EpilogueType{}(out_dram_window,
+                               o_acc_tile,
+                               ds_dram_windows,
+                               p_smem,
+                               context,
+                               std::forward<decltype(epilogue_args)>(epilogue_args)...);
+            },
+            args);
+    }
+
+    /// @brief Process epilogue with iteration index
+    template <typename ODramWindow,
+              typename OAccTile,
+              typename DsDramWindows,
+              typename Context,
+              index_t I>
+    CK_TILE_DEVICE void operator()(ODramWindow& out_dram_window,
+                                   const OAccTile& o_acc_tile,
+                                   const DsDramWindows& ds_dram_windows,
+                                   void* p_smem,
+                                   Context& context,
+                                   number<I> iAccess) const
+    {
+        ck_tile::apply(
+            [&](auto&&... epilogue_args) {
+                EpilogueType{}(out_dram_window,
+                               o_acc_tile,
+                               ds_dram_windows,
+                               p_smem,
+                               iAccess,
+                               context,
+                               std::forward<decltype(epilogue_args)>(epilogue_args)...);
+            },
+            args);
+    }
+};
+
+/// @brief Specialization for epilogue operation wrapper with no arguments
+template <typename EpilogueType>
+struct EpilogueNode<EpilogueType>
+{
+    using Epilogue = EpilogueType;
+    ck_tile::tuple<> args;
+
+    constexpr EpilogueNode() = default;
+
+    template <typename ODramWindow, typename OAccTile, typename DsDramWindows, typename Context>
+    CK_TILE_DEVICE void operator()(ODramWindow& out_dram_window,
+                                   const OAccTile& o_acc_tile,
+                                   const DsDramWindows& ds_dram_windows,
+                                   void* p_smem,
+                                   Context& context) const
+    {
+        EpilogueType{}(out_dram_window, o_acc_tile, ds_dram_windows, p_smem, context);
+    }
+
+    template <typename ODramWindow,
+              typename OAccTile,
+              typename DsDramWindows,
+              typename Context,
+              index_t I>
+    CK_TILE_DEVICE void operator()(ODramWindow& out_dram_window,
+                                   const OAccTile& o_acc_tile,
+                                   const DsDramWindows& ds_dram_windows,
+                                   void* p_smem,
+                                   Context& context,
+                                   number<I> iAccess) const
+    {
+        EpilogueType{}(out_dram_window, o_acc_tile, ds_dram_windows, p_smem, iAccess, context);
+    }
+};
```

Comment:
This isn't a generic epilogue node. It's highly tied up to the cshuffle epilogue functionality. I think we should change it, or at least for now maybe change the naming of the node to `CShuffleEpilogueNode`.